Automattic/wp-calypso
# Stats: Fix routes checking for Overview, Site, Summary, and WordAds

#### Proposed Changes

* ~Restore the `pathname` to compare paths without query string [as previous logic](https://github.com/Automattic/wp-calypso/blob/2436d081fb82fd0b18e25b0c5108033adc6e934d/client/my-sites/stats/controller.jsx#L191-L192).~
* Deprecate `altPaths` since there have been redirections handling the alternative paths.
* Extract `getWordAdsFilters` for the `Ads` page routes since `WordAds` [has its own routing](https://github.com/Automattic/wp-calypso/blob/24a58e16f34b1b78025956855a858ae57f2ea2f8/client/my-sites/stats/index.js#L73).

#### Testing Instructions



> This change also aims to make routes work in the [Stats Standalone App for Jetpack plugins](https://github.com/Automattic/wp-calypso/pull/69231).

* Spin up this change with the Calypso Live link.
* Ensure the following paths work.
* Overview routes:
    - `/stats` ( [redirected](https://github.com/Automattic/wp-calypso/blob/24a58e16f34b1b78025956855a858ae57f2ea2f8/client/my-sites/stats/index.js#L43) )
    - `/stats/day`
    - `/stats/week`
    - `/stats/month`
    - `/stats/year`
* Site routes:
    - `/stats/insights/${your-site}`
    - `/stats/${your-site}` ( [redirected](https://github.com/Automattic/wp-calypso/blob/24a58e16f34b1b78025956855a858ae57f2ea2f8/client/my-sites/stats/index.js#L80) )
    - `/stats/day/${your-site}`
    - `/stats/week/${your-site}`
    - `/stats/month/${your-site}`
    - `/stats/year/${your-site}`
* WordAds routes:
    - `/stats/ads/${your-site}` ( [redirected](https://github.com/Automattic/wp-calypso/blob/24a58e16f34b1b78025956855a858ae57f2ea2f8/client/my-sites/stats/index.js#L76-L77) )
    - `/stats/ads/day/${your-site}`
    - `/stats/ads/week/${your-site}`
    - `/stats/ads/month/${your-site}`
    - `/stats/ads/year/${your-site}`
* Summary routes (`posts` is one of the [valid modules](https://github.com/Automattic/wp-calypso/blob/24a58e16f34b1b78025956855a858ae57f2ea2f8/client/my-sites/stats/index.js#L28-L39)):
    - `/stats/posts/${your-site}` ( [redirected](https://github.com/Automattic/wp-calypso/blob/24a58e16f34b1b78025956855a858ae57f2ea2f8/client/my-sites/stats/index.js#L58) )
    - `/stats/day/posts/${your-site}`
    - `/stats/week/posts/${your-site}`
    - `/stats/month/posts/${your-site}`
    - `/stats/year/posts/${your-site}`

#### Pre-merge Checklist





Related to #70602
Related to #70563
